pkg/config/config.go

```diff
@@ -0,0 +1,84 @@
+package config
+
+// config package provides an API with which Navigator components can read yaml
+// or properties files and override certain keys before writing them back to the
+// filesystem.
+//
+// This is currently a thin wrapper around ``viper`` which is already able to read and
+// write files in multiple formats.
+// The ``Interface`` has only the operations which are needed by Navigator, so
+// we can swap viper something else if necessary in the future.
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	shutil "github.com/termie/go-shutil"
+)
+
+const (
+	BackupSuffix = ".navigator_original"
+)
+
+type config struct {
+	*viper.Viper
+}
+
+type Interface interface {
+	Get(key string) interface{}
+	Set(key string, value interface{})
+	Unset(key string)
+	WriteConfig() error
```

Comment:
Can this method be changed to accept an `io.Writer` as an argument?

The `New` function can also be changed to accept an `io.Read`, and we can then make the `newFrom` variants handle obtaining an `io.Reader` from a particular filepath.